packages/storybook-ui-components/stories/2-Icon.stories.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import { withKnobs, select } from "@storybook/addon-knobs";
+
+import { Icon } from "@cockroachlabs/ui-components";
+
+export default {
+  title: "Icon",
+  component: Icon,
+  decorators: [withKnobs],
+};
+
+const IconDisplaySection = ({ children }) => (
+  <section
+    style={{
+      display: "flex",
+      justifyContent: "flex-start",
+      flexFlow: "row wrap",
+    }}
+  >
+    {children}
+  </section>
+);
+
+const IconDisplay = ({ children, backgroundColor = "transparent" }) => (
+  <div
+    style={{
+      textAlign: "center",
+      backgroundColor: backgroundColor,
+      margin: "2rem 1rem",
+    }}
+  >
+    {children}
+  </div>
+);
+const IconFrame = ({ children }) => (
+  <div
+    style={{ display: "flex", height: "75px", width: "75px", margin: "auto" }}
+  >
+    {children}
+  </div>
+);
+
+const IconLabel = ({ text }) => (
+  <code style={{ fontFamily: "monospace", fontSize: "12px" }}>{text}</code>
+);
+
+const IconNames = [
+  "ArrowLeft",
+  "Backup",
+  "CancelCircleFilled",
+  "CancelCircle",
+  "Cancel",
+  "CaretDown",
+  "CaretFilledDown",
+  "CaretFilledRight",
+  "CaretLeft",
+  "CaretRight",
+  "CaretUp",
+  "Caution",
+  "CheckCircleFilled",
+  "CheckCircle",
+  "Check",
+  "CircleInfoFilled",
+  "Copy",
+  "Download",
+  "EllipsisVertical",
+  "Ellipsis",
+  "ErrorCirleFilled",
+  "ErrorCircle" ,
+  "EyeOff" ,
+  "Eye" ,
+  "InfoFilled" ,
+  "Info" ,
+  "Invalid" ,
+  "List" ,
+  "LockFilled" ,
+  "Lock",
+  "MinusCircle",
+  "Minus",
+  "Organization",
+  "Pencil",
+  "PlusCircle",
+  "Plus",
+  "Search",
+  "Spinner",
+  "Stack",
+  "Terminal",
+  "Time",
+  "User",
+  "World",
+];
+const sizes = ["large", "default", "small", "tiny"];
```

Comment:
Medium is missing

Reply:
It might be easier to keep these in sync if they are exported from the component.

Reply:
So I looked into this. I have exported the `IconSize` type, but since it is a [string literal type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types) it can not be iterated over like an array. I could export an array of strings of the literal type, but it wouldn't really serve any other purpose except in creating this story. I also considered an [Enum](https://www.typescriptlang.org/docs/handbook/enums.html) which could be iterated over, but would then require an implementer to import the Enum in order to pass props to the component an I don't think is as ergonomic as using a string.

So my compromise is to export the custom type and then type this array as `Array<IconSize>`. while this doesn't solve the problem of missing types, it does error if you try to add something to the array that's not of type IconSize. Since these stories aren't likely to change very often I don't think the maintenance cost is very high here.